### PR TITLE
Update changelog

### PR DESCRIPTION
**Pull request by aaronayres35**
This PR starts the process of updating the changelog in advance of the upcoming 7.2.0 release

It also pulls in the news fragment system previously introduced in apptools (see https://github.com/enthought/apptools/pull/181) to make the process of updating the changelog more efficient in future releases.

### CODE REVIEW

**Comment by rahulporuri** [comment]
@aaronayres35 looks like the only thing remaining on this PR are the detailed notes and highlights?

File: CHANGES.txt [hunk 1]
```diff
@@ -1,6 +1,147 @@
...
+~~~~~~~~~~~~~~~~~~
+
+* This release of TraitsUI now depends on Traits 6.1+. Also, deprecated code /
+  modules have been removed. Namely, the :mod:`traitsui.image <traitsui.image>`
+  module which was moved to ``pyface.image``, ``editors_gen`` modules, Editor
+  and EditorFactory factory methods on Toolkit objects, and more.  For a
+  complete list, see PRs in the "Removals" section below.  These were all
+  generally unused / deprecated for sometime. Also, importing directly from
+  ``traitsui.editors`` has been deprecated. Please update imports to import
+  directly from :mod:`traitsui.api` or :mod:`traitsui.editors.api`.
```
Comment (aaronayres35) [inline CHANGES.txt:29]: Which should we be recommending here? I listed both currently, but I imagine we just want people to import from `traitsui.api`?  
This has been discussed before

Comment (rahulporuri) [reply CHANGES.txt:29]: >  This has been discussed before

I dont remember that discussion - or the conclusion of that discussion.
I'm fine with listing both. I completely forgot about the fact that the editor factories are also importable via `traitsui.api`. I think we can leave it up to the users to choose. I don't think we'll remove `traitsui.editors.api`.

**Review by rahulporuri — APPROVED** [review]
LGTM

**Pull request merged**